server/internal/monitoring/monitoring.go

```diff
@@ -0,0 +1,54 @@
+package monitoring
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	metricNamespace = "llm_operator"
+
+	metricsNameCompletionLatency = "inference_manager_serever_completion_latency"
+)
+
+type MetricsMonitoring interface {
+	ObserveCompletionLatency(latency time.Duration)
+}
+
+// MetricsMonitor holds and updates Prometheus metrics.
+type MetricsMonitor struct {
+	completionLatencyHist prometheus.Histogram
+}
+
+// NewMetricsMonitor returns a new MetricsMonitor.
+func NewMetricsMonitor() *MetricsMonitor {
+	completionLatencyHist := prometheus.NewHistogram(
+		prometheus.HistogramOpts{
+			Namespace: metricNamespace,
+			Name:      metricsNameCompletionLatency,
+			// Exponential buckets up to 4 minutes -- 0.25 to 256.
+			Buckets: prometheus.ExponentialBuckets(0.25, 2, 11),
```

Comment:
We can do later, but we might want to add a label for a model ID specified in a request.